### PR TITLE
Update README for service registration

### PR DESCRIPTION
**Pull request by martincostello**
* Update the `README` with examples of using service registration from #49.
  * Add link to `CHANGELOG` for once the release is published.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -30,13 +30,56 @@ The concrete class that implements `IStatsDPublisher` is `StatsDPublisher`. The
...
+#### .NET Core
+
+An example of registering StatsD dependencies using `IServiceCollection`:
+
+```csharp
+services.AddStatsD(
+    (provider) =>
+    {
+        var options = provider.GetRequiredService<MyOptions>().StatsD;
+        var logger = provider.GetRequiredService<ILogger<StatsDConfiguration>>();
```
Comment (AnthonySteele) [inline README.md]: Logger is not used in this example. Perahs should be in the `LogError(ex)` line?

Comment (martincostello) [reply README.md]: Good spot. It used to be, but then I took the code from the existing example and simplified it.

File: README.md [hunk 2]
```diff
@@ -30,13 +30,55 @@ The concrete class that implements `IStatsDPublisher` is `StatsDPublisher`. The
...
+{
+    public MyService(IStatsDPublisher statsPublisher)
+    {
+        // Use the publisher
+    }
+}
+
+```
+
+#### .NET 4.5.1
```
Comment (AnthonySteele) [inline README.md]: You can still do this .Net core. You just aren't taking advantage of the service registration bits.

Comment (martincostello) [reply README.md]: Suggested alternate heading?

Comment (AnthonySteele) [reply README.md]: Hmmm, maybe

#### Simplest example
No service registration or Ioc. Works in .NET 4.5.1 and .NET core.

**Pull request merged**